src/ngraph/op/util/op_annotations.hpp

```diff
@@ -56,6 +57,17 @@ namespace ngraph
 
                 bool is_cacheable() const { return m_cacheable; }
                 void set_cacheable(bool val) { m_cacheable = val; }
+                /// \returns a deep copy of this OpAnnotations object.
+                virtual std::shared_ptr<OpAnnotations> clone()
+                {
+                    auto* clone = new OpAnnotations(*this);
```

Comment:
> Why not just return std::make_shared<OpAnnotations>(*this) ?
Because the copy constructor was intentionally made protected so that we only use `clone` to create copies, taking into account the specific object of the class hierarchy. For that reason, we cannot do `std::make_shared<OpAnnotations>(*this)`.

> This also looks like a shallow copy to me.
At this point, shallow copy and deep copy are equivalent in this class but that might change in the future if we add more members to it.

Does it make sense?